### PR TITLE
Change MessageAttachment type, stubs for json api

### PR DESCRIPTION
**Pull request by patrickxb**
I meant for this to just have the change in chat_local.avdl to MessageAttachment, but some more stuff snuck in too, which is in progress work on integrating attachments into `keybase chat api`.

Also had to do some lint cleanup to commit.

I'll add some comments.

### CODE REVIEW

File: go/client/cmd_chat_api.go [hunk 1]
```diff
@@ -371,6 +371,12 @@ func (c *CmdChatAPI) EditV1(ctx context.Context, opts editOptionsV1) Reply {
 	return c.sendV1(ctx, arg)
 }
 
+// AttachV1 implements ChatServiceHandler.AttachV1.
+func (c *CmdChatAPI) AttachV1(ctx context.Context, opts attachOptionsV1) Reply {
+	// TODO: implement
```
Comment (patrickxb) [inline go/client/cmd_chat_api.go]: WIP: and this will get implemented.

File: go/protocol/chat1/local.go [hunk 2]
```diff
@@ -27,56 +27,14 @@ type MessageDelete struct {
...
-var MessageAttachmentVersionRevMap = map[MessageAttachmentVersion]string{
-	1: "V1",
-}
-
-type MessageAttachmentV1 struct {
-	Path string `codec:"path" json:"path"`
-	Key  []byte `codec:"key" json:"key"`
-}
-
 type MessageAttachment struct {
```
Comment (patrickxb) [inline go/protocol/chat1/local.go]: This is the main change up for review.

**Comment by patrickxb** [comment]
cc @oconnor663

File: protocol/avdl/chat1/local.avdl [hunk 3]
```diff
@@ -21,22 +21,14 @@ protocol local {
...
-  variant MessageAttachment switch (MessageAttachmentVersion version ) {
-    case V1: MessageAttachmentV1;   
+  record MessageAttachment {
+    string path;               // path to the attachment object
+    int size;                  // size of the attachment object (bytes)
+    string previewPath;        // path to the preview object (optional)
+    int previewSize;           // size of the preview object (required if previewPath set)
+    string mimeType;           // mime type of the attachment
+    bytes metadata;            // generic metadata (msgpack)
+    bytes key;                 // encryption key
```
Comment (songgao) [inline protocol/avdl/chat1/local.avdl]: Are we using different key(s) for attachment vs messages? Do different attachments use different keys? What was the reason for this?

(These are not really review comments but I'm just interested to know!)

Comment (patrickxb) [inline protocol/avdl/chat1/local.avdl]: Yes different key. S3 can essentially be public. User has to get key from chat message to decrypt.  

And the benefits jack mentions for delete.

Comment (maxtaco) [inline protocol/avdl/chat1/local.avdl]: I would prefer an `Asset` object that has a (path,size) pair and then have two instances of it, one for the object, and one (with a `*` for the preview object).

Comment (patrickxb) [inline protocol/avdl/chat1/local.avdl]: @maxtaco PTAL

File: go/protocol/chat1/local.go [hunk 4]
```diff
@@ -27,16 +27,18 @@ type MessageDelete struct {
...
-	PreviewSize    int    `codec:"previewSize" json:"previewSize"`
-	MimeType       string `codec:"mimeType" json:"mimeType"`
-	Metadata       []byte `codec:"metadata" json:"metadata"`
-	Key            []byte `codec:"key" json:"key"`
-	AttachmentHash Hash   `codec:"attachmentHash" json:"attachmentHash"`
-	PreviewHash    Hash   `codec:"previewHash" json:"previewHash"`
+	Object   Asset  `codec:"object" json:"object"`
+	Preview  *Asset `codec:"preview,omitempty" json:"preview,omitempty"`
+	Metadata []byte `codec:"metadata" json:"metadata"`
+	Key      []byte `codec:"key" json:"key"`
```
Comment (maxtaco) [inline go/protocol/chat1/local.go]: I'd move the `Key` into the `Asset` too, otherwise we'll have to be fancier with the nonces.

**Pull request merged**